### PR TITLE
Add native skip

### LINKED ISSUE #1690: Skip is not honored when set to true
### Steps to reproduce

```dart
void main() {

  patrolTest('dfsdf', skip: true, ($) async {
      print('Hello world');
  });
```

### Actual results

When running above code via `patrol test` the test does not get skipped.
The app stays open in the simulator and the process keeps running until you manually kill it.

### Logs

```
Test Suite 'All tests' started at 2023-09-14 12:56:50.190
        Test Suite 'RunnerUITests.xctest' started at 2023-09-14 12:56:50.192
        Test Suite 'RunnerUITests' started at 2023-09-14 12:56:50.192
        Test Case '-[RunnerUITests tests_features_prepayment_updatePrepaymentTest]' started.
            t =     0.00s Start Test at 2023-09-14 12:56:50.195
            t =     0.03s Set Up
            t =     0.03s Open xx
            t =     0.05s     Launch xx
            t =     0.06s         Terminate xx:25724
            t =     1.35s         Setting up automation session
            t =     2.54s         Wait for xx to idle
        2023-09-14 12:56:54.254616+0200 RunnerUITests-Runner[25559:7696005] PatrolServer: INFO: App reported that it is ready
        2023-09-14 12:56:54.660292+0200 RunnerUITests-Runner[25559:7695803] PatrolAppServiceClient.runDartTest(tests.features...)
```

### Patrol version

v2.1.3

### Patrol Doctor output

```
Patrol CLI version: 2.1.3
Program adb found in /opt/homebrew/bin/adb
Env var $ANDROID_HOME set to /Users/cambiph/Library/Android/sdk
Program xcodebuild found in /usr/bin/xcodebuild
Program ideviceinstaller found in /opt/homebrew/bin/ideviceinstaller
```

### Flutter Doctor output

```Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.13.1, on macOS 13.5.2 22G91 darwin-arm64, locale en-BE)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[!] Android Studio (version unknown)
    ✗ Unable to determine Android Studio version.
[✓] Android Studio (version 2022.2)
[✓] IntelliJ IDEA Community Edition (version 2023.1.2)
[✓] VS Code (version 1.82.0)
[✓] VS Code (version 1.82.1)
[✓] Connected device (3 available)
[✓] Network resources

! Doctor found issues in 1 category.
```

### PR DESCRIPTION
**Pull request by pdenert**
fix #1690

### CODE REVIEW

File: packages/patrol/example/android/app/src/androidTest/java/pl/leancode/patrol/example/MainActivityTest.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package pl.leancode.patrol.example;
+package pl.leancode.patrol.e2e_app;
```
Comment (piotruela) [inline packages/patrol/example/android/app/src/androidTest/java/pl/leancode/patrol/example/MainActivityTest.java:1]: Is that change intended?

File: packages/patrol/example/android/app/src/androidTest/java/pl/leancode/patrol/example/MainActivityTest.java [hunk 2]
```diff
@@ -17,11 +17,11 @@ public static Object[] testCases() {
         return instrumentation.listDartTests();
     }
 
-    public MainActivityTest(String dartTestName) {
+    public MainActivityTest(String dartTestCaseNamesList) {
         this.dartTestName = dartTestName;
     }
 
-    private final String dartTestName;
+    private String dartTestName;
```
Comment (piotruela) [inline packages/patrol/example/android/app/src/androidTest/java/pl/leancode/patrol/example/MainActivityTest.java:24]: How about those two?

File: packages/patrol/example/android/app/src/androidTest/java/pl/leancode/patrol/example/MainActivityTest.java (hunk 1 above)
Comment (pdenert) [reply packages/patrol/example/android/app/src/androidTest/java/pl/leancode/patrol/example/MainActivityTest.java:1]: No

**Review by piotruela — APPROVED** [review]

**Review by mateuszwojtczak — APPROVED** [review]
